Source/Core/InputCommon/ControllerInterface/DInput/DInputKeyboardMouse.h

```diff
@@ -106,11 +116,36 @@ class KeyboardMouse : public Core::Device
 
 private:
   void UpdateCursorInput();
+  void Generate_Octagon_Points(POINT octagon_points[8]);
+  bool Point_Is_Inside_Octagon(const POINT& mouse_point, const POINT octagon_points[8]);
+  double Calculate_Distance_between_Points(const POINT& first_point, const POINT& second_point);
+  long Find_Second_Line_Point(const POINT& mouse_point, const POINT octagon_points[8],
+                              long index_of_min_octagon_point);
+  void Move_Mouse_Point_Along_Gate(POINT& mouse_point, const POINT octagon_points[8]);
+  void Lock_Mouse_In_Jail(POINT& mouse_point);
+
+  bool player_requested_mouse_center = false;
+  const double screen_height = static_cast<double>(GetSystemMetrics(SM_CYFULLSCREEN));
+  const double screen_width = static_cast<double>(GetSystemMetrics(SM_CXFULLSCREEN));
+  const double screen_ratio = screen_width / screen_height;
+  const POINT center_of_screen =
+      POINT{static_cast<long>(screen_width / 2.0), static_cast<long>(screen_height / 2.0)};
+  enum octagon_points
+  {
+    SOUTH,
+    SOUTH_EAST,
+    EAST,
+    NORTH_EAST,
+    NORTH,
+    NORTH_WEST,
+    WEST,
+    SOUTH_WEST
+  };
 
   const LPDIRECTINPUTDEVICE8 m_kb_device;
   const LPDIRECTINPUTDEVICE8 m_mo_device;
 
   DWORD m_last_update;
-  State m_state_in;
+  State current_state;
```

Comment:
```suggestion
  State m_current_state;
```

Reply:
I really hate these decorators. My thinking goes: if it's in the class, we know it's a member variable. There can't be a context where we can access that variable where it's not attached to a Keyboard Mouse object. 

However, I'm happy to change this if it's a mandatory part of the Dolphin standard. I modified it originally because I was under the impression that the decorators (Hungarian notation?) were optional and we were just allowed to use g_, m_, and s_ if we wanted but those decorators were not mandatory. Are we required to use those decorators?

Reply:
Yes, it's required by the Dolphin coding style.

I think the reasoning is that when you're reading a function, you can easily tell which variables are local variables, which are member variables (here I am thinking of members of `this`), which are static and which are global – without having to loop up where the variable is defined. Certainly there is also an argument to be made for not using these prefixes, but for the sake of consistency we should do it the same way in all of Dolphin.

Reply:
`m_` is also useful since it makes it harder to accidentally shadow the member variables

Reply:
Roger. I will update my variables then.